### PR TITLE
Adding mangrove toggle to all layers

### PR DESCRIPTION
**Pull request by gislawill**
https://www.loom.com/share/0f6bc235544d4916b7fd73c020b25c2e?sid=b95dd979-24e7-4389-9641-aaf980bec9d1

### CODE REVIEW

File: src/Map.js [hunk 1]
```diff
@@ -250,13 +253,16 @@ export default function Map() {
...
+          <LegendLayerSwitch
+            layersToggle={layersToggle}
+            toggleLayer={toggleLayer}
+            layerGroup={layerGroup}
+          />
+          <LegendLayerToggle
+            layersToggle={layersToggle}
+            toggleLayer={toggleLayer}
+            layerGroup={layerGroup}
+          />
```
Comment (gislawill) [inline src/Map.js:265]: Splitting this into two to keep each component simpler

File: src/hooks/layers/useLayers.js [hunk 2]
```diff
@@ -50,12 +50,12 @@ export function useLayers(
...
-      Object.keys(availableLayers).reduce((acc, layer) => {
-        const key = availableLayers[layer]?.sharedKey ?? layer;
-        if (availableLayers[layer]?.sharedKey) {
+    setLayersToggle({
+      ...Object.keys(availableLayers).reduce((acc, layer) => {
+        const key = availableLayers[layer]?.toggleKey ?? layer;
+        if (availableLayers[layer]?.toggleKey) {
           const keys = Object.keys(availableLayers).filter(
-            (l) => availableLayers[l]?.sharedKey === key,
+            (l) => availableLayers[l]?.toggleKey === key,
```
Comment (gislawill) [inline src/hooks/layers/useLayers.js:58]: Reminded sharedKey to toggleKey to denote what it does

File: src/hooks/layers/useLayers.js [hunk 3]
```diff
@@ -70,15 +70,16 @@ export function useLayers(
         }
         return acc;
       }, {}),
-    );
+      ...layersToggle,
```
Comment (gislawill) [inline src/hooks/layers/useLayers.js:73]: This keeps the mangrove layer selection as you jump between layers

File: src/hooks/layers/useLayers.js [hunk 4]
```diff
@@ -136,10 +137,32 @@ export function useLayers(
...
+                layerToggleToUse[associatedLayerName] &&
+                (toggleKey
+                  ? layerToggleToUse[toggleKey]
+                  : layerToggleToUse[layer.key])
+              );
+            } else {
+              return toggleKey
+                ? layerToggleToUse[toggleKey] ?? layerToggleToUse[layer.key]
+                : layerToggleToUse[layer.key];
+            }
```
Comment (gislawill) [inline src/hooks/layers/useLayers.js:165]: Logic to decide whether or not to render the layer based on layer config and toggles

File: src/layers/protos/FillWithOutlineProto.js [hunk 5]
```diff
@@ -35,12 +35,6 @@ export default class FillWithOutlineProto {
     this.opacity = opacity;
     this.minzoom = minzoom;
     this.maxzoom = maxzoom;
-    this.visible =
-      (subgroup !== undefined && subgroup === floodGroup) ||
-      subgroup === undefined ||
-      floodGroup === undefined
-        ? "visible"
-        : "none";
```
Comment (gislawill) [inline src/layers/protos/FillWithOutlineProto.js:43]: Remove this logic, we only add layers to the map that we actually want to view

File: src/legends/legend-layer-selector.tsx [hunk 6]
```diff
@@ -171,8 +136,87 @@ export default memo(function LegendLayerSelector({
           )}
         </div>
       ))}
+    </div>
+  );
+};
 
-      {Object.entries(sharedKeyLayers).map(
+export const LegendLayerToggle = ({
```
Comment (gislawill) [inline src/legends/legend-layer-selector.tsx:143]: Separated this out into it's own component for simplicity and positioning

**Comment by lowriech** [comment]
THIS IS WONDERFUL

**Pull request merged**